specification/compute/resource-manager/Microsoft.Compute/stable/2021-03-01/cloudService.json

```diff
@@ -2116,6 +2462,184 @@
+        "id": {
+          "description": "Resource Id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Resource location.",
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/OSVersionProperties"
+        }
+      }
+    },
+    "OSVersionListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OSVersion"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "OSVersionPropertiesBase": {
+      "description": "Configuration view of an OS version.",
+      "type": "object",
+      "properties": {
+        "version": {
+          "description": "The OS version.",
+          "type": "string",
+          "readOnly": true
+        },
+        "label": {
+          "description": "The OS version label.",
+          "type": "string",
+          "readOnly": true
+        },
+        "isDefault": {
+          "description": "Specifies whether this is the default OS version for its family.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "isActive": {
+          "description": "Specifies whether this OS version is active.",
+          "type": "boolean",
+          "readOnly": true
+        }
+      }
+    },
+    "OSFamilyProperties": {
+      "description": "OS family properties.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The OS family name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "label": {
+          "description": "The OS family label.",
+          "type": "string",
+          "readOnly": true
+        },
+        "versions": {
+          "description": "List of OS versions belonging to this family.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OSVersionPropertiesBase"
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "OSFamily": {
+      "description": "Describes a cloud service OS family.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource Id.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
```

Comment:
Similar to other location resource types in compute, I guess locations/cloudServiceOsFamilies resource also uses ProxyOnly, LocationBased routing. Property 'location' cannot be used as a top level property for proxy resources.
Refer the NOTE section in the RP documentation here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

Reply:
I see that the note says that location is not applicable in the request body. However, these ProxyOnly resources are only available through GET. Additionally readonly: true is specified. So location would only be applied for the response body in this situation.

Reply:
As per offline discussion, as cloudServiceOsFamilies can differ based on a location, including location in response sounds reasonable.